### PR TITLE
Add Transcode API

### PR DESCRIPTION
**Pull request by leszko**
Adds testing for the Studio API `/api/transcode`.

Two scenarios are tested:
- Transcoding a file from a publicly available URL
- Transcoding a file from a private Storj bucket

Note that I added this the same way as VOD, so it's a continuous test, but I think we should consider running it more like acceptance test + smoke test. I don't think we need to run it because of the MediaConvert cost.

---------------------

**To productize it we'll need to**
- Create a Livepeer Testing Storj bucket
- Add it as a job (continuously running pod) to our k8 cluster
- (optionally) Consider cleaning up the bucket from time to time

-------------

**Notes to reviewers**
- All functional changes are included in one commit e88766bf84655cfc8729461fc14fc08b8bed2ffb
- Two other commits are refactoring, they improve code reusability, but decrease the encapsulation a little bit
- This related PR https://github.com/livepeer/go-api-client/pull/23 needs to be merged first and `go.mod` dep updated

### CODE REVIEW

File: internal/app/common/continuous_tester.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+			messenger.SendMessage(fmt.Sprintf("Continuous test of %s on %s cancelled", ct.name, ct.host))
+			return ct.ctx.Err()
+		} else if ctxErr != nil {
+			msg := fmt.Sprintf("Test of %s on %s timed out, potential deadlock! ctxErr=%q err=%q", ct.name, ct.host, ctxErr, err)
+			messenger.SendFatalMessage(msg)
+		} else if err != nil || es != 0 {
+			msg := fmt.Sprintf(":rotating_light: Test of %s on %s ended with err=%v errCode=%v", ct.name, ct.host, err, es)
+			messenger.SendFatalMessage(msg)
+			glog.Warning(msg)
+			ct.sendPagerdutyEvent(err)
```
Comment (iameli) [inline internal/app/common/continuous_tester.go:78]: I know you're probably following existing patterns in stream-tester, but I've been really trying to drive everything toward Grafana. Centralizing our alerting there  allows everything to be managed in one place, allows alerts to be tuned and silenced when necessary — very useful.

File: internal/app/common/continuous_tester.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+	"github.com/PagerDuty/go-pagerduty"
+
+	"github.com/golang/glog"
+	"github.com/livepeer/stream-tester/messenger"
+)
+
+type (
+	IContinuousTester interface {
+		// Start test. Blocks until error.
+		Start(start func(ctx context.Context) (int, error), testDuration, pauseBetweenTests time.Duration) error
```
Comment (thomshutt) [inline internal/app/common/continuous_tester.go:18]: What's the thinking behind having an error code and an error object returned here?

File: internal/app/common/continuous_tester.go (hunk 1 above)
Comment (leszko) [reply internal/app/common/continuous_tester.go:78]: I think centralizing everything with Grafana is a good idea. Though, I think we should change it in a separate PR. What do you think?

This PR is just to add the Transcode API support in the exactly the same way we support VOD API.

Btw. do you have any example of using Grafana in streamtester?

File: internal/app/common/continuous_tester.go (hunk 2 above)
Comment (leszko) [reply internal/app/common/continuous_tester.go]: To be honest I don't know, I think it was copy paster from recordtester into the vod tester. I just refactored something we already had in VOD tester 🙃 

Saying that, when I looked again in the code, I think we return always `0` or `1` for the error code, so I think it's safe to remove. Then, removed.

**Review by thomshutt — APPROVED** [review]

File: internal/app/common/continuous_tester.go (hunk 1 above)
Comment (leszko) [reply internal/app/common/continuous_tester.go:78]: Ok, merging this PR. We can do the refactoring to use Grafana (for both VOD and Storj) in a separate PR.

**Pull request merged**